.github/workflows/rust-mirai.yml

```diff
@@ -0,0 +1,64 @@
+name: Rust
+
+on:
+  pull_request:
+    types: ['opened', 'edited', 'reopened', 'synchronize']
```

Comment:
I don't think we need to rerun the analysis on edited and reopened. Is this common to another CI focused GitHub actions?

Reply:
list of types is [here](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#pull-request-event-pull_request) but it's not clear to me what is everything. I think you are right though, any changes to the PR will be reflected by `synchronize`.